From 25a620f2e952cdbc10b148f403013e07e0b7285f Mon Sep 17 00:00:00 2001
Message-Id: <25a620f2e952cdbc10b148f403013e07e0b7285f.1418766606.git.jen@redhat.com>
In-Reply-To: <6f81b4847eb68ebdf54a8f1a771e19d112d74152.1418766606.git.jen@redhat.com>
References: <6f81b4847eb68ebdf54a8f1a771e19d112d74152.1418766606.git.jen@redhat.com>
From: Fam Zheng <famz@redhat.com>
Date: Thu, 4 Dec 2014 00:05:04 -0600
Subject: [CHANGE 10/31] qemu-iotests: Don't run 005 on vmdk split formats
To: rhvirt-patches@redhat.com,
    jen@redhat.com

RH-Author: Fam Zheng <famz@redhat.com>
Message-id: <1417651524-18041-11-git-send-email-famz@redhat.com>
Patchwork-id: 62683
O-Subject: [RHEL-7.1 qemu-kvm PATCH v5 10/30] qemu-iotests: Don't run 005 on vmdk split formats
Bugzilla: 1002493
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Max Reitz <mreitz@redhat.com>

There would be too many extents that VMDK driver can't open all of them:

 >  005 0s ... - output mismatch (see 005.out.bad)
 >  --- 005.out     2013-12-24 09:27:27.608181030 +0800
 >  +++ 005.out.bad 2014-02-13 10:00:15.282184557 +0800
 >  @@ -4,10 +4,10 @@
 >   Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=5368709120000
 >
 >   small read
 >  -read 4096/4096 bytes at offset 1024
 >  -4 KiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
 >  +qemu-io: can't open device /tmp/qemu-iotests/t.vmdk: Could not open '/tmp/qemu-iotests/t-s1016.vmdk': Too many open files
 >  +no file open, try 'help open'
 >
 >   small write
 >  -wrote 4096/4096 bytes at offset 8192
 >  -4 KiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
 >  +qemu-io: can't open device /tmp/qemu-iotests/t.vmdk: Could not open '/tmp/qemu-iotests/t-s1016.vmdk': Too many open files
 >  +no file open, try 'help open'
 >   *** done

So disable the two subformats.

Signed-off-by: Fam Zheng <famz@redhat.com>
Reviewed-by: Kevin Wolf <kwolf@redhat.com>
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
(cherry picked from commit 2aa4a86f59c7093e03d2ec18c5d5f08c957d1a78)
Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Jeff E. Nelson <jen@redhat.com>
---
 tests/qemu-iotests/005 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/qemu-iotests/005 b/tests/qemu-iotests/005
index 9abcb84..ba1236d 100755
--- a/tests/qemu-iotests/005
+++ b/tests/qemu-iotests/005
@@ -44,6 +44,8 @@ trap "_cleanup; exit \$status" 0 1 2 3 15
 _supported_fmt generic
 _supported_proto generic
 _supported_os Linux
+_unsupported_imgopts "subformat=twoGbMaxExtentFlat" \
+                     "subformat=twoGbMaxExtentSparse"
 
 # vpc is limited to 127GB, so we can't test it here
 if [ "$IMGFMT" = "vpc" ]; then
-- 
2.1.0

